server/app/views/applicant/ProgramIndexView.java

```diff
@@ -516,18 +527,42 @@ private LiTag programCard(
             "aria-label",
             messages.at(
                 buttonSrText.getKeyName(), program.localizedName().getOrDefault(preferredLocale)))
-        .withClasses(ReferenceClasses.APPLY_BUTTON, ButtonStyles.SOLID_BLUE_TEXT_SM, "mx-auto");
+        .withClasses(ReferenceClasses.APPLY_BUTTON, ButtonStyles.SOLID_BLUE_TEXT_SM);
 
-    DivTag actionDiv = div(content).withClasses("w-full", "mb-6", "flex-grow", "flex", "items-end");
-    return li().withId(baseId)
-        .withClasses(ReferenceClasses.APPLICATION_CARD, ApplicantStyles.PROGRAM_CARD)
-        .with(
-            // The visual bar at the top of each program card.
-            div()
-                .withClasses(
-                    "block", "shrink-0", BaseStyles.BG_SEATTLE_BLUE, "rounded-t-xl", "h-3"))
-        .with(programData)
-        .with(actionDiv);
+    DivTag actionDiv =
+        div(content).withClasses("mt-4", "mb-6", "flex", "items-center", "justify-center");
+    LiTag finalDiv =
+        li().withId(baseId)
+            .withClasses(ReferenceClasses.APPLICATION_CARD, ApplicantStyles.PROGRAM_CARD)
+            .with(
+                // The visual bar at the top of each program card.
+                div()
+                    .withClasses(
+                        "block", "shrink-0", BaseStyles.BG_SEATTLE_BLUE, "rounded-t-xl", "h-3"));
+    programImage.ifPresent(finalDiv::with);
+
+    return finalDiv.with(programData).with(actionDiv);
+  }
+
+  private Optional<ImgTag> createProgramImage(Http.Request request, ProgramDefinition program) {
+    if (!settingsManifest.getProgramCardImages(request)) {
+      return Optional.empty();
+    }
+    if (program.summaryImageFileKey().isEmpty()) {
+      return Optional.empty();
+    }
+    String summaryImageFileKey = program.summaryImageFileKey().get();
+    if (!PublicFileNameFormatter.isFileKeyForPublicProgramImage(summaryImageFileKey)) {
+      return Optional.empty();
+    }
+    // TODO(#5676): Can we detect if the image URL is invalid and then not show it?
+    // TODO(#5676): Include a placeholder while the image is loading.
+
+    return Optional.of(
+        img()
+            .withSrc(publicStorageClient.getPublicDisplayUrl(program.summaryImageFileKey().get()))
```

Comment:
suggestion (blocking):  This doesn't handle the `RuntimeException` that can be thrown further up the stack in `AwsStorageUtils.localStackActionLink()`.  I'm thinking that either we should handle the exception in `localStackActionLink()` and maybe return an empty string as the url and log a warning.  Or we could catch the exception here and return `Optional.empty()`.

Reply:
Looks good!